src/bpf/bpfctl.c

```diff
@@ -0,0 +1,238 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <getopt.h>
+
+#include "build.h"
+#include "format-table.h"
+#include "format-util.h"
+#include "main-func.h"
+#include "parse-argument.h"
+#include "pretty-print.h"
+#include "varlink.h"
+#include "verbs.h"
+
+static JsonFormatFlags arg_json_format_flags = JSON_FORMAT_OFF;
+
+typedef struct BpfProgramData BpfProgramData;
+typedef struct BpfMapData BpfMapData;
+
+struct BpfProgramData {
+        char *name;
+        uint64_t type;
+        uint64_t memlock;
+        char *tag;
+        char **map_names;
+        uint64_t run_time_ns;
+        uint64_t run_cnt;
+};
+
+struct BpfMapData {
+        char *name;
+        uint64_t type;
+        uint64_t memlock;
+};
+
+static void bpf_prog_data_done(BpfProgramData *p) {
+        assert(p);
+
+        p->name = mfree(p->name);
+        p->tag = mfree(p->tag);
+        strv_free(p->map_names);
+}
+
+static void bpf_map_data_done(BpfMapData *p) {
+        assert(p);
+
+        p->name = mfree(p->name);
+}
+
+static int query_bpf_progs_and_maps(void) {
+        _cleanup_(varlink_unrefp) Varlink *vl = NULL;
+        _cleanup_(table_unrefp) Table *map_table = NULL, *prog_table = NULL;
+        JsonVariant *i = NULL, *o = NULL, *m = NULL, *v = NULL;
+        int r;
+
+        map_table = table_new("map_name", "map_type", "memlock");
+        if (!map_table)
+                return log_oom();
+
+        prog_table = table_new("program_name", "program_type", "memlock", "program_tag", "program_map_names", "run_time_ns", "run_cnt");
```

Comment:
as above, drop the redundant "program_" prefix, and use spaces rather than under scores.

goal is short human friendly names, that are not too cryptic